### ISSUE TITLE
Wrong documentation and bug with body streaming

### ISSUE THREAD

**Issue by marcadella** [playframework/playframework#10937]
### Play Version

2.8.7

### API

Scala

### Operating System

4.15.0-142-generic

### JDK

openjdk version "11.0.10" 2021-01-19

### Expected Behavior

The doc provides an example of how to stream a body in a POST request: https://www.playframework.com/documentation/2.8.x/ScalaBodyParsers#Directing-the-body-elsewhere

Firstly, `ws.url(...)` sets the method to `GET`. So the body parser needs to call `withMethod("POST")` to fix that.

I modified the example like this:
```scala
def forward(request: WSRequest): BodyParser[WSResponse] = BodyParser { req =>
    Accumulator.source[ByteString].mapFuture { source =>
      request
        .withBody(source) 
        .withMethod("POST") //<-
        .execute()
        .map(Right.apply)
    }
  }
```

and I expect to see the body sent to the server.

### Actual Behavior

... But the server receives the request without the body. It looks like the source is not consumed at all.

### Reproducible Test Case

https://www.playframework.com/documentation/2.8.x/ScalaBodyParsers#Directing-the-body-elsewhere

**Comment by mkurz** [comment]
@marcadella Thanks! Do you want to provide a pull request to fix the docs?

**Comment by marcadella** [comment]
There was two things: the doc error, but also the fact that the server receives the request without body (which is the main issue).

**Comment by mkurz** [comment]
@marcadella I fixed the documentation here: #11195

However, your claim

> ... But the server receives the request without the body. It looks like the source is not consumed at all.

isn't true IMHO. The body gets consumed and will be forwarded to the target of `ws.url(...)`, like written [in the docs](https://www.playframework.com/documentation/2.8.x/ScalaBodyParsers#Directing-the-body-elsewhere): "...when you actually don’t want to parse the body, rather, **you want to stream it elsewhere.**"

I created a reproducer: https://github.com/mkurz/playframework-reproducers/tree/main/playframework-10937 so you can try yourself. Please have a look at the README in that repo, it describes how to run the app(s). As you can see two apps get  started and to one we send a POST request with a body and that app will redirect the body to the second app which prints the headers and the body to the console. Everything works.

**Comment by mkurz** [comment]
@marcadella Merged the doc fix pr #11195.
I am closing this issue since my reproducer in the above commit works. Please let me know if you think I am wrong.
